### PR TITLE
Add a simple populate script

### PR DESCRIPTION
**Pull request by nsarrazin**
### Why

When testing changes I found myself often having to generate a bunch of data manually to get started testing. This PR adds a script that we can run to create users, assistants and conversations. 

### How to use
You can populate the database using faker data using the `populate` script:

```bash
npm run populate <flags here>
```

At least one flag must be specified, the following flags are available:

- `reset` - resets the database
- `users` - populates the users table
- `settings` - populates the settings table for users in the db
- `assistants` - populates the assistants table for users in the db
- `conversations` - populates the conversations table for users in the db

> [!WARNING]  
> The `MONGODB_URL` used for this script will be fetched from `.env.local`. Make sure it's correct! The command runs directly on the database.

For example, you could use it like so:

```bash
npm run populate reset
```

to clear out the database. Then login in the app to create your user in the DB, and run the following command:

```bash
npm run populate users settings assistants conversations
```

to populate the database with fake data, including fake conversations and assistants for your user.

### Mock data

| image  | image |
| ------------- | ------------- |

### Missing stuff 
(Not a prio to add, unless we need it for testing)
- websearch messages
- Support for non-logged in users (we can have settings and conversations linked to anonymous sessions)
- Avatar for assistants

### CODE REVIEW

**Comment by nsarrazin** [comment]
could be useful for testing search related changes in the future @mishig25 :grin:

**Comment by everypizza1** [comment]
> ### Why
> When testing changes I found myself often having to generate a bunch of data manually to get started testing. This PR adds a script that we can run to create users, assistants and conversations.
> 
> ### How to use
> You can populate the database using faker data using the `populate` script:
> 
> ```shell
> npm run populate <flags here>
> ```
> 
> At least one flag must be specified, the following flags are available:
> 
> * `reset` - resets the database
> * `users` - populates the users table
> * `settings` - populates the settings table for users in the db
> * `assistants` - populates the assistants table for users in the db
> * `conversations` - populates the conversations table for users in the db
> 
> Warning
> 
> The `MONGODB_URL` used for this script will be fetched from `.env.local`. Make sure it's correct! The command runs directly on the database.
> 
> For example, you could use it like so:
> 
> ```shell
> npm run populate reset
> ```
> 
> to clear out the database. Then login in the app to create your user in the DB, and run the following command:
> 
> ```shell
> npm run populate users settings assistants conversations
> ```
> 
> to populate the database with fake data, including fake conversations and assistants for your user.
> 
> ### Mock data
> image	image
> ### Missing stuff
> (Not a prio to add, unless we need it for testing)
> 
> * websearch messages
> * Support for non-logged in users (we can have settings and conversations linked to anonymous sessions)
> * Avatar for assistants

File: README.md [hunk 1]
```diff
@@ -616,3 +616,36 @@ npm run updateLocalEnv
...
+
+You can populate the database using faker data using the `populate` script:
+
+```bash
+npm run populate <flags here>
+```
+
+At least one flag must be specified, the following flags are available:
+
+- `reset` - resets the database
```
Comment (mishig25) [inline README.md:630]: can we have flag `all` that does all of them ?

File: README.md [hunk 2]
```diff
@@ -616,3 +616,36 @@ npm run updateLocalEnv
...
+- `users` - populates the users table
+- `settings` - populates the settings table for existing users
+- `assistants` - populates the assistants table for existing users
+- `conversations` - populates the conversations table for existing users
+
+> [!WARNING]  
+> The `MONGODB_URL` used for this script will be fetched from `.env.local`. Make sure it's correct! The command runs directly on the database.
```
Comment (mishig25) [inline README.md:637]: this section should be placed right after `### Populate database`

File: scripts/populate.ts [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+async function seed() {
+	console.log("Seeding...");
+	const modelIds = models.map((model) => model.id);
+
+	if (flags.includes("reset")) {
+		await collections.users.deleteMany({});
+		await collections.settings.deleteMany({});
+		await collections.assistants.deleteMany({});
+		await collections.conversations.deleteMany({});
+		console.log("Reset done");
```
Comment (mishig25) [inline scripts/populate.ts:113]: we should have similar `console.log("Reset started...");` and console statements for other types as well: `users`, `assistants` (like `console.log("Users created. Added 100 new users");`)

**Comment by mishig25** [comment]
Also add a sentence in https://github.com/huggingface/chat-ui?tab=readme-ov-file#launch saying "To populate the database, go [here](#populate-database)"

File: README.md (hunk 2 above)
Comment (nsarrazin) [reply README.md]: Agreed, done!

**Comment by nsarrazin** [comment]
Implemented the review feedback! I think I want to keep it in the docs at the bottom, under the HuggingChat section as it's more a tool for developers than for users. wdyt? For the rest should be all good

**Review by mishig25 — APPROVED** [review]
lgtm !

**Pull request merged**